app/Controller/UsersController.php

```diff
@@ -31,6 +31,9 @@ public function beforeFilter()
 
         // what pages are allowed for non-logged-in users
         $allowedActions = array('login', 'logout');
+        if(!empty(Configure::read('Security.email_otp_enabled'))) {
+          $allowedActions[] = 'email_otp';
```

Comment:
should this not be below the self-registration?
As self_registration should work independently of email_otp